adapters/Mintegral/MintegralAdapter/waterfall/GADMAdapterMintegralNativeAdLoader.m

```diff
@@ -0,0 +1,285 @@
+    [_adEventDelegate reportImpression];
+}
+
+#pragma mark - GADMediatedUnifiedNativeAd
+- (NSString *)headline {
+    return _campaign.appName;
+}
+
+- (NSArray *)images {
+    return nil;
+}
+
+- (NSString *)body {
+    return _campaign.appDesc;
+}
+
+- (GADNativeAdImage *)icon {
+    return _icon;
+}
+
+- (NSString *)callToAction {
+    return _campaign.adCall;
+}
+
+- (NSDecimalNumber *)starRating {
+    NSString *star = [NSString stringWithFormat:@"%@",[_campaign valueForKey:@"star"]];
+    return [NSDecimalNumber decimalNumberWithString:star];
+}
+
+- (NSString *)store {
+    return nil;
+}
+
+- (NSString *)price {
+    return nil;
+}
+
+-(NSString *)advertiser{
+    return nil;
+}
+
+- (NSDictionary *)extraAssets {
+    return nil;
+}
+
+- (BOOL)hasVideoContent {
+    return YES;
+}
+
+-  (UIView *)mediaView{
+    [_mediaView setMediaSourceWithCampaign:_campaign unitId:_adUnitId];
+    return _mediaView;
+}
+
+- (UIView *)adChoicesView {
+    return _adChoicesView;
+}
+
+#pragma mark - GADMediationNativeAd
+- (BOOL)handlesUserClicks{
+    return YES;
+}
+
+- (BOOL)handlesUserImpressions{
+    return YES;
+}
+
+#pragma mark - GADMediatedUnifiedNativeAd
+-(void)didRenderInView:(UIView *)view clickableAssetViews:(NSDictionary<GADNativeAssetIdentifier,UIView *> *)clickableAssetViews nonclickableAssetViews:(NSDictionary<GADNativeAssetIdentifier,UIView *> *)nonclickableAssetViews viewController:(UIViewController *)viewController{
+    [_nativeManager registerViewForInteraction:view withClickableViews:clickableAssetViews.allValues withCampaign:_campaign];
+}
+
+- (void)loadRequiredNativeData {
+    GADMAdapterMintegralNativeAdLoader *__weak weakSelf = self;
+    void (^localBlock)(void) = ^{
+        GADMAdapterMintegralNativeAdLoader *strongSelf = weakSelf;
+        if (strongSelf && strongSelf->_adLoadCompletionHandler) {
+            strongSelf->_adEventDelegate = strongSelf->_adLoadCompletionHandler(strongSelf, nil);
+        }
+    };
+    NSString *URLString = _campaign.iconUrl;
+    if (!URLString.length) {
+        localBlock();
+        return;
+    }
+    NSURL *URL = [NSURL URLWithString:URLString];
+    if (!URL) {
+        localBlock();
+        return;
+    }
+    NSURLSession *session = [NSURLSession sharedSession];
+    NSURLSessionDataTask *task =
+    [session dataTaskWithURL:URL
+           completionHandler:^(NSData *_Nullable data, NSURLResponse *_Nullable response,
+                               NSError *_Nullable error) {
+        dispatch_async(dispatch_get_main_queue(), ^{
+            GADMAdapterMintegralNativeAdLoader *strongSelf = weakSelf;
+            if (!strongSelf) {
+                return;
+            }
+            GADNativeAdImage *image =
+            (!error && data)
+            ? [[GADNativeAdImage alloc] initWithImage:[UIImage imageWithData:data]]
+            : nil;
+            strongSelf->_icon = image;
+            localBlock();
+        });
+    }];
+    [task resume];
+}
```

Comment:
Nit, seems like the implementation of this method remain the same in both waterfall and bidding, so we can consider moving this to the utils as well.

Reply:
ok,I have modified.